src/web/components/StickyBottomBanner/StickyBottomBanner.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import {
+    CMP,
+    shouldShow as shouldShowCMP,
+} from '@root/src/web/components/StickyBottomBanner/CMP';
+import { ReaderRevenueBanner } from '@root/src/web/components/StickyBottomBanner/ReaderRevenueBanner';
+
+type Props = {
+    isSignedIn?: boolean;
+    countryCode?: string;
+    CAPI: CAPIBrowserType;
+};
+
+export const StickyBottomBanner = ({
+    isSignedIn,
+    countryCode,
+    CAPI,
+}: Props) => {
+    const [showCMP, setShowCMP] = useState<boolean | null>(null);
+
+    useEffect(() => {
+        const callShouldShow = () => setShowCMP(shouldShowCMP());
+
+        if (CAPI.config.cmpUi) {
+            callShouldShow();
+        }
+    }, [CAPI.config.cmpUi]);
+
+    // Don't render anything until we know whether we can show the CMP
+    if (showCMP === null) {
```

Comment:
I found this a bit confusing but I don't know of a better pattern here. Could expess intent even clearer by adding an `isUnset` method or something which just does the `=== null` check, but that is probably overkill.

Reply:
Yeah, and maybe this becomes a bit more obvious/rigorous if we introduce some "banner picking" functionality to support extra banners OR preferably we defer all decision making to automat.